simd/src/Kokkos_SIMD.hpp

```diff
@@ -193,7 +193,7 @@ using host_abi_set = abi_set<simd_abi::scalar, simd_abi::avx2_fixed_size<4>,
                              simd_abi::avx2_fixed_size<8>>;
 using data_type_set =
     data_types<std::int32_t, std::int64_t, std::uint64_t, double, float>;
-#elif defined(KOKKOS_ARCH_ARM_NEON)
+#elif defined(KOKKOS_ARCH_ARM_NEON)  && !defined(__NVCC__)
```

Comment:
We handled a similar problem with `nvc++` slightly differently: https://github.com/kokkos/kokkos/blob/7d0e3e8184779c6f6b0d3ad7eac0341dbf273a2c/cmake/kokkos_arch.cmake#L532-L535

Reply:
@masterleinad I was thinking of that before but it seems Christian started to add some checks to `KOKKOS_SIMD.hpp` already, so I added the last missing check. Should I remove the checks and disbale neon in cmake?

Reply:
It should be consistent at the very least. I would prefer disabling it in CMake for now.